### PR TITLE
Prevent to self-close if dangerouslySetInnerHTML set

### PR DESCRIPTION
**Pull request by clomie**
This change prevents to self-close if the element has `dangerouslySetInnerHTML` prop. This is required for better xml support.

Currently behavior:
```
import { h } from 'preact'
import renderToString from 'preact-render-to-string'

console.log(renderToString(<a dangerouslySetInnerHTML={{ __html: '<b />' }}></a>, undefined, { xml: true }))
// <a><b / />

console.log(renderToString(<link dangerouslySetInnerHTML={{ __html: '<foo />' }}></link>))
// <link><foo / />
```

### CODE REVIEW

**Review by sventschui — APPROVED** [review]
Thank you very much for your contribution! This fix looks good to me, just waiting for another approver before merging.

**Review by JoviDeCroock — APPROVED** [review]
Looks great, thank you!

**Pull request merged**
